pkg/gardenlet/controller/shoot/shoot/reconciler_migrate.go

```diff
@@ -250,18 +264,33 @@ func (r *Reconciler) runMigrateShootFlow(ctx context.Context, o *operation.Opera
 			Dependencies: flow.NewTaskIDs(deleteStaleExtensionResources),
 		})
 		migrateControlPlane = g.Add(flow.Task{
-			Name:         "Migrating shoot control plane",
-			Fn:           botanist.Shoot.Components.Extensions.ControlPlane.Migrate,
+			Name: "Migrating shoot control plane",
+			Fn: flow.TaskFn(func(ctx context.Context) error {
+				if isWorkerless {
```

Comment:
`SkipIf` can't be used at many places because, while generating the graph of the flow it will panic at many place, like below, ref - https://github.com/gardener/gardener/blob/46372035b02a80748aa43e3449678da9d9477016/pkg/gardenlet/controller/shoot/shoot/reconciler_reconcile.go#L322-L331, Here since `controlplane` is nil for workerless there is no wait function for it and if we use skipif it will panic.
There are places we can use `skipif` for workerless but for harmonizing reason we have kept it similar everywhere.

Related to other comment that in `gardenlet` logs will still be there for things which are being deployed, yes it will be there, for that we have thought of enhancing flow so that a task can be added based on condition. But that can be done as part of separate PRs.

 I will close all other similar comment for `skipif` or `doif` for now, so that it can be easy for us to address open comments, we can continue the discussion here.